wire-ios/Wire-iOS/Sources/UserInterface/Folders/Mappers/Folder+LabelType.swift

```diff
@@ -17,10 +17,14 @@
 //
 
 import WireDataModel
+import WireMoveToFolderUI
 
-public protocol CreateConversationFolderUseCaseProtocol {
-    /// Creates a new conversation folder with the specified name
-    /// - Parameter name: The name of the folder to be created
-    /// - Returns: The created `LabelType`, or `nil` if creation fails
-    func invoke(with name: String) async -> LabelType?
+extension Folder {
+
+    init(_ label: LabelType) {
+        self.init(
+            identifier: label.remoteIdentifier,
+            name: label.name ?? ""
```

Comment:
This case kind of makes no sense right where there is no name. It is kind of propagating an invalid state. Do you just use it in `CreateConversationFolderUseCase`. Consider deleting this extension and initializing with id & name. In `CreateConversationFolderUseCase` you can throw an error if the name is nil.

Reply:
Simplified it by removing the extension